docs/content/connectors/oracle-cdc.md

```diff
@@ -0,0 +1,472 @@
+   
+    (1.3). Check whether log archiving is enabled
+    ```sql
+    -- Should now "Database log mode: Archive Mode"
+    archive log list;
+    ```
+   <b>Notes:</b>
+   
+   Supplemental logging must be enabled for captured tables or the database in order for data changes to capture the <em>before</em> state of changed database rows.
+   The following illustrates how to configure this on the table/database level.
+   ```sql
+   -- Enable supplemental logging for a specific table:
+   ALTER TABLE inventory.customers ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS;
+   ```
+   ```sql
+   -- Enable supplemental logging for database
+   ALTER DATABASE ADD SUPPLEMENTAL LOG DATA;
+   ```
+
+2. Create an Oracle user with permissions
+
+   (2.1). Create Tablespace
+   ```sql
+   sqlplus sys/password@host:port/SID AS SYSDBA;
+     CREATE TABLESPACE logminer_tbs DATAFILE '/opt/oracle/oradata/SID/logminer_tbs.dbf' SIZE 25M REUSE AUTOEXTEND ON MAXSIZE UNLIMITED;
+     exit;
+   ```
+
+   (2.2). Create a user and grant permissions
+   ```sql
+   sqlplus sys/password@host:port/SID AS SYSDBA;
+     CREATE USER flinkuser IDENTIFIED BY flinkpw DEFAULT TABLESPACE LOGMINER_TBS QUOTA UNLIMITED ON LOGMINER_TBS;
+     GRANT CREATE SESSION TO flinkuser;
+     GRANT SET CONTAINER TO flinkuser;
+     GRANT SELECT ON V_$DATABASE to flinkuser;
+     GRANT FLASHBACK ANY TABLE TO flinkuser;
+     GRANT SELECT ANY TABLE TO flinkuser;
+     GRANT SELECT_CATALOG_ROLE TO flinkuser;
+     GRANT EXECUTE_CATALOG_ROLE TO flinkuser;
+     GRANT SELECT ANY TRANSACTION TO flinkuser;
+     GRANT LOGMINING TO flinkuser;
+
+     GRANT CREATE TABLE TO flinkuser;
+     GRANT LOCK ANY TABLE TO flinkuser;
+     GRANT ALTER ANY TABLE TO flinkuser;
+     GRANT CREATE SEQUENCE TO flinkuser;
+
+     GRANT EXECUTE ON DBMS_LOGMNR TO flinkuser;
+     GRANT EXECUTE ON DBMS_LOGMNR_D TO flinkuser;
+
+     GRANT SELECT ON V_$LOG TO flinkuser;
+     GRANT SELECT ON V_$LOG_HISTORY TO flinkuser;
+     GRANT SELECT ON V_$LOGMNR_LOGS TO flinkuser;
+     GRANT SELECT ON V_$LOGMNR_CONTENTS TO flinkuser;
+     GRANT SELECT ON V_$LOGMNR_PARAMETERS TO flinkuser;
+     GRANT SELECT ON V_$LOGFILE TO flinkuser;
+     GRANT SELECT ON V_$ARCHIVED_LOG TO flinkuser;
+     GRANT SELECT ON V_$ARCHIVE_DEST_STATUS TO flinkuser;
+     exit;
+   ```
+
+See more about the [Setting up Oracle](https://debezium.io/documentation/reference/1.5/connectors/oracle.html#setting-up-oracle)
+
+
+How to create an Oracle CDC table
+----------------
+
+The Oracle CDC table can be defined as following:
+
+```sql 
+-- register an Oracle table 'products' in Flink SQL
+Flink SQL> CREATE TABLE products (
+     id INT NOT NULL,
+     name STRING,
+     description STRING,
+     weight DECIMAL(10, 3),
+     PRIMARY KEY(id) NOT ENFORCED
+     ) WITH (
+     'connector' = 'oracle-cdc',
+     'hostname' = 'localhost',
+     'port' = '1521',
+     'username' = 'flinkuser',
+     'password' = 'flinkpw',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)